src/conf_mode/high-availability.py

```diff
@@ -40,33 +40,41 @@ def get_config(config=None):
     else:
         conf = Config()
 
-    base = ['high-availability', 'vrrp']
+    base = ['high-availability']
+    base_vrrp = ['high-availability', 'vrrp']
     if not conf.exists(base):
         return None
 
-    vrrp = conf.get_config_dict(base, key_mangling=('-', '_'),
+    ha = conf.get_config_dict(base, key_mangling=('-', '_'),
                                 get_first_key=True, no_tag_node_value_mangle=True)
     # We have gathered the dict representation of the CLI, but there are default
     # options which we need to update into the dictionary retrived.
-    if 'group' in vrrp:
-        default_values = defaults(base + ['group'])
-        for group in vrrp['group']:
-            vrrp['group'][group] = dict_merge(default_values, vrrp['group'][group])
+    if 'vrrp' in ha:
+        if 'group' in ha['vrrp']:
+            default_values_vrrp = defaults(base_vrrp + ['group'])
+            for group in ha['vrrp']['group']:
+                ha['vrrp']['group'][group] = dict_merge(default_values_vrrp, ha['vrrp']['group'][group])
+
+    # Merge per virtual-server default values
+    if 'virtual_server' in ha:
+        default_values = defaults(base + ['virtual-server'])
+        for vs in ha['virtual_server']:
+            ha['virtual_server'][vs] = dict_merge(default_values, ha['virtual_server'][vs])
 
     ## Get the sync group used for conntrack-sync
     conntrack_path = ['service', 'conntrack-sync', 'failover-mechanism', 'vrrp', 'sync-group']
     if conf.exists(conntrack_path):
-        vrrp['conntrack_sync_group'] = conf.return_value(conntrack_path)
+        ha['conntrack_sync_group'] = conf.return_value(conntrack_path)
 
-    return vrrp
+    return ha
 
-def verify(vrrp):
-    if not vrrp:
+def verify(ha):
+    if not ha:
         return None
 
     used_vrid_if = []
-    if 'group' in vrrp:
-        for group, group_config in vrrp['group'].items():
+    if 'group' in ha:
```

Comment:
Shouldnt this be: if 'vrrp' in ha and 'group' in ha['vrrp']:?

Reply:
> Shouldnt this be: if 'vrrp' in ha and 'group' in ha['vrrp']:?

Sure, fixed